packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -73,13 +73,18 @@ function createHttpClient(): HttpClient {
 
 export function getCacheVersion(
   paths: string[],
-  compressionMethod?: CompressionMethod
+  compressionMethod?: CompressionMethod,
+  crossOsEnabled = false
 ): string {
-  const components = paths.concat(
-    !compressionMethod || compressionMethod === CompressionMethod.Gzip
-      ? []
-      : [compressionMethod]
-  )
+  const components = paths
+    .concat(
+      !compressionMethod || compressionMethod === CompressionMethod.Gzip
+        ? []
+        : [compressionMethod]
```

Comment:
Should we remove this as we are not doing any special casing for gzip?

Reply:
It will cause cache invalidation, but that will get masked by invalidation due to `windows-only`

Reply:
I think this is there so that cache created by gzip is only restorable by gzip. If self-hosted runners don't have zstd then it falls back to gzip. 
We can remove this if we callout that having zstd in path is a requirement for cache to work

Reply:
Then in that case we can remove gzip from our code. And that seems like a big change

Reply:
Won't the fallback logic take care of this?

Reply:
We need different versions of cache for different compression algorithms as same compression algo should be used to compress(save) and decompress(restore)

Reply:
@Phantsure yes, and that is happening as we are putting `[compressionMethod]` in the version (L83)

Reply:
@bishal-pdMSFT Yes, but you were asking if we should remove it. Right? We need to keep some component of compression algo in version

Reply:
So just to understand, if we remove this statement, cache invalidation will happen because the newer versions will include `gzip` in their version hash and that won't match for older gzip caches. So even in this case, shouldn't gzip caches only be restorable by using gzip. Previously they were identified by not having `gzip` in version but moving forward will have `gzip` in version. I think this is what @bishal-pdMSFT is referring to.

Reply:
>Previously they were identified by not having gzip in version but moving forward will have gzip in version.

Yes, this is what I meant. We need not put empty compression method here for gzip and let it be handled just like any other compression method.

Reply:
> Previously they were identified by not having gzip in version but moving forward will have gzip in version.

Then we can do that. Remove empty compression case here.

Reply:
Empty string still remains if no compression method is passed. Removed gzip case. This was present since initial commit. Let me check if it is really needed